app/src/pages/inside/stepPage/modals/testItemDetailsModal/testItemDetailsModal.jsx

```diff
@@ -44,105 +56,221 @@ const messages = defineMessages({
     const {
       intl,
       data: { item },
+      userAccountRole,
+      userProjectRole,
+      userId,
+      handleSubmit,
     } = this.props;
     const okButton = {
-      text: intl.formatMessage(COMMON_LOCALE_KEYS.OK),
-      onClick: (closeModal) => closeModal(),
+      text: intl.formatMessage(COMMON_LOCALE_KEYS.SAVE),
+      onClick: (closeModal) => {
+        handleSubmit(this.updateItemAndCloseModal(closeModal))();
+      },
+    };
+    const cancelButton = {
+      text: intl.formatMessage(COMMON_LOCALE_KEYS.CANCEL),
     };
+
+    const editable = canEditLaunch(
+      userAccountRole,
+      userProjectRole,
+      item.owner ? userId === item.owner : userId === item.owner,
+    );
     return (
-      <ModalLayout title={intl.formatMessage(messages.modalTitle)} okButton={okButton}>
-        <div className={cx('name')}>{item.name}</div>
-        <div className={cx('row')}>
-          <LabeledSection label={intl.formatMessage(messages.testCaseId)}>
-            {item.uniqueId}
-          </LabeledSection>
+      <ModalLayout
+        title={intl.formatMessage(messages.modalTitle)}
+        okButton={editable ? okButton : undefined}
+        cancelButton={cancelButton}
+        closeConfirmation={editable ? this.getCloseConfirmationConfig() : undefined}
+        warningMessage={editable ? intl.formatMessage(messages.launchWarning) : ''}
+      >
+        <div className={cx('name-row')}>
+          <div className={cx('name')}>{item.name}</div>
+          <TestItemStatus status={item.status} />
         </div>
-        <div className={cx('row')}>
-          <LabeledSection label={intl.formatMessage(messages.duration)}>
-            {getDuration(item.startTime, item.endTime)}
-          </LabeledSection>
-          {item.tags &&
-            item.tags.length > 0 && (
-              <LabeledSection className={cx('tags')} label={intl.formatMessage(messages.tags)}>
-                {item.tags.map((tag) => (
-                  <span key={tag} className={cx('tag')}>
-                    {tag}
-                  </span>
-                ))}
-              </LabeledSection>
+        <ModalField label={intl.formatMessage(messages.testCaseId)}>
+          <div className={cx('id')}>{item.uniqueId}</div>
+        </ModalField>
+        <ModalField label={intl.formatMessage(messages.duration)}>
+          {getDuration(item.startTime, item.endTime)}
+        </ModalField>
+        <ModalField label={intl.formatMessage(messages.location)}>
+          <div className={cx('location')} title={item.location}>
+            {item.location}
+            {item.location && (
+              <CopyToClipboard text={item.location} className={cx('copy')}>
+                {Parser(IconDuplicate)}
+              </CopyToClipboard>
             )}
-        </div>
-        {item.parameters && (
-          <div className={cx('row')}>
-            <TestParameters parameters={item.parameters} />
           </div>
+        </ModalField>
+        {item.tags &&
```

Comment:
Tags are not used anymore, so please remove this code (they were replaced with attributes)